src/vs/editor/common/languages/autoIndent.ts

```diff
@@ -60,6 +60,47 @@ function getPrecedingValidLine(model: IVirtualModel, lineNumber: number, process
 	return -1;
 }
 
+/**
+ * Get indentation for line from reference line
+ */
+function getIndentationForLineFromReferenceLine(
+	autoIndent: EditorAutoIndentStrategy,
+	model: IVirtualModel,
+	languageConfigurationService: ILanguageConfigurationService,
+	considerOnEnterRulesForInheritedIndentAfterBlankLine: boolean,
+	indentConverter: IIndentConverter | undefined,
+	referencePrevLineNum: number,
+	currentLineNumber: number
+): string {
+	let indentation = strings.getLeadingWhitespace(model.getLineContent(referencePrevLineNum));
+	// Check for onEnter rules that should decrease the indent
+	if (indentConverter && considerOnEnterRulesForInheritedIndentAfterBlankLine) {
+		const richEditSupport = languageConfigurationService.getLanguageConfiguration(model.tokenization.getLanguageId());
+		if (richEditSupport) {
+			const previousLineText = referencePrevLineNum < 1 ? '' : model.getLineContent(referencePrevLineNum - 1);
+			let afterEnterText = "";
+			try {
+				afterEnterText = model.getLineContent(currentLineNumber);
+			} catch (e) {
+				// Probably after the end of file
+			}
```

Comment:
This was an issue if the last line of the file is `if (1)` and you hit enter. I feel like there's probably a better way to do this, but couldn't find an easy was to see if it was passed the end of the file.